jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java

```diff
@@ -13,25 +13,196 @@
 
-import java.util.List;
 
 public class WebSocketDiagnosticsCollector implements DiagnosticsCollector {
-    public WebSocketDiagnosticsCollector() {
-    }
+	public WebSocketDiagnosticsCollector() {
+	}
+
+	private Diagnostic createDiagnostic(IJavaElement el, ICompilationUnit unit, String msg, String code) {
+		try {
+			ISourceRange nameRange = JDTUtils.getNameRange(el);
+			Range range = JDTUtils.toRange(unit, nameRange.getOffset(), nameRange.getLength());
+			Diagnostic diagnostic = new Diagnostic(range, msg);
+			diagnostic.setCode(code);
+			completeDiagnostic(diagnostic);
+			return diagnostic;
+		} catch (JavaModelException e) {
+			JakartaCorePlugin.logException(WebSocketConstants.DIAGNOSTIC_ERR_MSG, e);
+		}
+		return null;
+	}
+
+	@Override
+	public void completeDiagnostic(Diagnostic diagnostic) {
+		diagnostic.setSource(WebSocketConstants.DIAGNOSTIC_SOURCE);
+		diagnostic.setSeverity(WebSocketConstants.ERROR);
+	}
+
+	@Override
+	public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnostics) {
+		if (unit == null) {
+			return;
+		}
+
+		IType[] alltypes;
+		IField[] allFields;
+		IAnnotation[] allFieldAnnotations;
+		IMethod[] allMethods;
+		IAnnotation[] allMethodAnnotations;
+
+		HashMap<String, Boolean> checkWSEnd = null;
+
+		try {
+			alltypes = unit.getAllTypes();
+			for (IType type : alltypes) {
+				checkWSEnd = isWSEndpoint(type);
+
+				// checks if the class uses annotation to create a WebSocket endpoint
+				if (checkWSEnd.get("isAnnotation")) {
+					invalidParamsCheck(type, WebSocketConstants.ON_OPEN, WebSocketConstants.ON_OPEN_PARAM_OPT_TYPES, 
+					        WebSocketConstants.RAW_ON_OPEN_PARAM_OPT_TYPES, unit, diagnostics);
+				}
+
+			}
+		} catch (JavaModelException e) {
+			JakartaCorePlugin.logException(WebSocketConstants.DIAGNOSTIC_ERR_MSG, e);
+		}
+	}
+
+	private void invalidParamsCheck(IType type, String methodAnnotTarget, Set<String> specialParamTypes, Set<String> rawSpecialParamTypes, ICompilationUnit unit,
+			List<Diagnostic> diagnostics) throws JavaModelException {
+
+		IMethod[] allMethods = type.getMethods();
+
+		for (IMethod method : allMethods) {
+			IAnnotation[] allAnnotations = method.getAnnotations();
+
+			for (IAnnotation annotation : allAnnotations) {
+				if (annotation.getElementName().equals(methodAnnotTarget)) {
+					ILocalVariable[] allParams = method.getParameters();
+
+					for (ILocalVariable param : allParams) {
+					    
+					    String signature = param.getTypeSignature();
+					    String formatSignature = signature.replace("/", ".");
+					    IType primaryType = param.getTypeRoot().findPrimaryType();
+					    String resolvedTypeName = JavaModelUtil.getResolvedTypeName(formatSignature, primaryType);
+					    					    
+					    boolean isPrimitive = JavaModelUtil.isPrimitive(formatSignature);
+					    boolean isSpecialType = specialParamTypes.contains(resolvedTypeName);
```

Comment:
probably makes more sense to initialize this to false. Otherwise line 122 is redundant

Reply:
yeah, I missed that part. I will fix that, thanks!